flink-libraries/flink-table/src/test/java/org/apache/flink/table/runtime/batch/sql/JavaSqlITCase.java

```diff
@@ -73,6 +73,30 @@ public void testValues() throws Exception {
 		compareResultAsText(results, expected);
 	}
 
+	@Test
+	public void testValuesWithCast() throws Exception {
```

Comment:
Can you move this test to `org.apache.flink.table.runtime.batch.sql.SetOperatorsITCase` and also add one to `org.apache.flink.table.runtime.stream.sql.SetOperatorsITCase`?

In addition it would be good to have to plan tests for this query in `org.apache.flink.table.api.batch.sql.SetOperatorsTest` and `org.apache.flink.table.api.stream.sql.SetOperatorsTest`.